woocommerce/wc-admin
# Only allow Customer Effort Score gathering if tracking is allowed

Fixes #5339

This adds a condition such that if tracking is not allowed, the CES hook JS won't be added to the output. It also adds a similar condition to the `CustomerEffortScoreTracks` JS component.

### Detailed test instructions:

- With tracking allowed (`/wp-admin/admin.php?page=wc-settings&tab=advanced&section=woocommerce_com`), and with `delete from wp_options where option_name = 'woocommerce_ces_shown_for_actions'` executed
- Add a new product. The CES modal should be displayed.
- Add another product. The CES modal should _not_ be displayed.
- Disable tracking and `delete from wp_options where option_name = 'woocommerce_ces_shown_for_actions'` again
- Add another product. The CES modal should _not_ be displayed.

To test the JS component, comment out the `return` in `/src/Features/CustomerEffortScoreTracks.php:61`:

```
		// Only add the JS if tracking is allowed.
		$allow_tracking = 'yes' === get_option( 'woocommerce_allow_tracking', 'no' );
		if ( ! $allow_tracking ) {
			//return;            <------ here
		}
```

### Changelog Note:

Dev: Only allow Customer Effort Score gathering if tracking is allowed
